docker/scripts/entrypoint.sh

```diff
@@ -20,92 +30,51 @@ NH_SECRET_PHRASE_PATH=${NH_SECRET_PHRASE_PATH:-"/data/secret_phrase.dat"}
 echo "NH_SECRET_PHRASE_PATH=${NH_SECRET_PHRASE_PATH}"
 
 # Node configurations (env->arg)
-NH_CONF_NAME=${NH_CONF_NAME:-"MyNode"}
-NH_CONF_BASE_PATH=${NH_CONF_BASE_PATH:-"/data/node"}
-NH_CONF_CHAIN=${NH_CONF_CHAIN:-"test"}
-NH_CONF_VALIDATOR=${NH_CONF_VALIDATOR:-}
-NH_CONF_NODE_KEY_FILE=${NH_CONF_NODE_KEY_FILE:-}
-NH_CONF_BOOTNODES=${NH_CONF_BOOTNODES:-}
-NH_CONF_RPC_CORS=${NH_CONF_RPC_CORS:-}
-NH_CONF_RPC_EXTERNAL=${NH_CONF_RPC_EXTERNAL:-}
-NH_CONF_RPC_METHODS=${NH_CONF_RPC_METHODS:-}
-NH_CONF_PRUNING=${NH_CONF_PRUNING:-}
-NH_PROMETHEUS_EXTERNAL=${NH_PROMETHEUS_EXTERNAL:-}
-
-for var_name in NH_CONF_NAME NH_CONF_BASE_PATH NH_CONF_CHAIN NH_CONF_VALIDATOR NH_CONF_NODE_KEY_FILE NH_CONF_BOOTNODES NH_CONF_RPC_CORS NH_CONF_RPC_EXTERNAL NH_CONF_RPC_METHODS NH_CONF_PRUNING NH_PROMETHEUS_EXTERNAL; do
-  # Get the value of the variable
-  var_value="${!var_name}"
-
-  # Check if the variable is defined
-  if [ -n "${var_value}" ]; then
-    echo "${var_name}=${var_value}"
-  else
-    echo "${var_name} is empty"
+prefix="NH_CONF_"
+conf_args=""
+echo "Node configuration:"
+while IFS='=' read -r -d '' var_name var_value; do
+  if [[ "$var_name" == NH_CONF* ]]; then
```

Comment:
Is it possible to use `${prefix}` here?

Reply:
Yes, should be allowed.

```
prefix="NH_CONF_"
var_name="NH_CONF_BASE_PATH"

if [[ "${var_name}" == ${prefix}* ]]; then
  echo "FITS"
fi
```

Reply:
Yes I agree, actually it was that way (`${prefix}*`) but an unintended undo slipped inside :blush:.